### PR TITLE
[Do Not Merge]Add repeatable fieldset support

### PR DESCRIPTION
**Pull request by Pareesh**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://loanconsolidate--nedbank--hlxsites.hlx.page/en/personal/borrow/personal-loans-form1
- After: https://loanconsolidate1--nedbank--hlxsites.hlx.page/en/personal/borrow/personal-loans-form1

### CODE REVIEW

File: blocks/form/decorators/fieldsets.js [hunk 1]
```diff
@@ -1,12 +1,83 @@
...
+  return item;
+}
+
 export default function decorateFieldsets(fieldsets, form) {
   form.querySelectorAll('fieldset').forEach((fieldsetEl) => {
     const fields = form.querySelectorAll(fieldsets[fieldsetEl.name]);
     if (fields.length) {
-      fields[0].insertAdjacentElement('beforebegin', fieldsetEl);
-      fieldsetEl.append(...fields);
+      if (fieldsetEl.hasAttribute('repeatable')) {
```
Comment (vdua) [inline blocks/form/decorators/fieldsets.js:62]: can we have the template as `fieldsetEl.outerHTML` ?

File: blocks/form/decorators/fieldsets.js [hunk 2]
```diff
@@ -1,12 +1,83 @@
...
+      item = event.target.closest('fieldset');
+      fieldset = item.parentElement.closest('fieldset');
+      item.remove();
+    }
+    fieldset.count += eventName === 'added' ? 1 : -1;
+    update(fieldset);
+    fieldset.dispatchEvent(new CustomEvent(`form-fieldset-item:${eventName}`, { detail: { item }, bubbles: true }));
+    event.stopPropagation();
+  };
+  return button;
```
Comment (vdua) [inline blocks/form/decorators/fieldsets.js:48]: the markup of this button is different than the button we create in the form block ? Can we fix that

Comment (vdua) [reply blocks/form/decorators/fieldsets.js:48]: I think in the case of repeatable fieldset, the form block should add two buttons add/remove on its own.

File: blocks/form/decorators/fieldsets.js [hunk 3]
```diff
@@ -1,12 +1,83 @@
...
+
+function update(fieldset) {
+  fieldset[Number(fieldset.max || -1) <= fieldset.count ? 'setAttribute' : 'removeAttribute']('maxxed', '');
+  const queue = [fieldset];
+  for (let i = 0; i < queue.length; i += 1) {
+    Array.from(queue[i].children).filter((c) => c.tagName === 'FIELDSET').forEach((item, index) => {
+      updateLegend(item, index);
+      item.name = `${queue[i].name}_${index + 1}`;
+      Array.from(item.elements)
+        .filter((c) => ['INPUT', 'SELECT', 'TEXTAREA', 'FIELDSET'].includes(c.tagName)).forEach((el) => {
```
Comment (vdua) [inline blocks/form/decorators/fieldsets.js:14]: tolowercase

**Review by vdua — CHANGES_REQUESTED** [review]
* fix different markup for button here than elsewhere in the form
* id generation logic should be same
* name of the field should be same so that you can get array of elements when you are doing operations on that paricular field
* templating logic can be simplified

File: blocks/form/decorators/fieldsets.js [hunk 4]
```diff
@@ -1,12 +1,83 @@
+function updateLegend(fieldsetItem, i) {
+  const legend = fieldsetItem.querySelector('legend');
+  legend.childNodes[0].data = legend.dataset.textTemplate.replace('#', i + 1);
```
Comment (vdua) [inline blocks/form/decorators/fieldsets.js:3]: I was thinking of doing it via css counters.

File: blocks/form/decorators/fieldsets.js (hunk 2 above)
Comment (Pareesh) [reply blocks/form/decorators/fieldsets.js:48]: Hi @vdua ,
For Nedbank case, a single Add button is shown for whole repeatable fieldset and one remove button for fieldsetitem. I have kept the implementation to satisfy this case. In future maybe we can append code for Add button per fieldset item.

File: blocks/form/decorators/repayments-calculator.js [hunk 5]
```diff
@@ -21,6 +21,8 @@ const groups = {
 
 const fieldsets = {
   insuranceOptionFieldSet: ['insuranceOption'].map(getSelector),
+  loanFieldSet: ['loanType', 'loanAmount'].map(getSelector),
+  extraCashFieldSet: ['extraCashHeading', 'extraCash'].map(getSelector),
```
[the remark this answers is not in the document]
Comment (Pareesh) [reply blocks/form/decorators/repayments-calculator.js:56]: yes

File: blocks/form/decorators/fieldsets.js [hunk 6]
```diff
@@ -1,12 +1,81 @@
-export default function decorateFieldsets(fieldsets, form) {
+import { nameToId } from '../form.js';
+
+function update(fieldset) {
+  const queue = [fieldset];
+  for (let i = 0; i < queue.length; i += 1) {
+    [...queue[i].children].filter((c) => c.tagName === 'FIELDSET').forEach((item, index) => {
+      const legend = item.querySelector('legend');
+      legend.childNodes[0].data = legend.dataset.textTemplate.replace('#', index + 1);
```
Comment (Pareesh) [inline blocks/form/decorators/fieldsets.js:8]: Implement using css counters

**Pull request closed without merging**